lib/AL_USDMaya/AL/usdmaya/nodes/ProxyDrawOverride.cpp

```diff
@@ -397,7 +397,16 @@ void ProxyDrawOverride::draw(const MHWRender::MDrawContext& context, const MUser
       ptr->m_engine->RenderBatch(combined, params);
     }
 
-    ptr->m_engine->Render(ptr->m_rootPrim, ptr->m_params);
+    glPushAttrib(GL_COLOR_BUFFER_BIT);
```

Comment:
glPushAttrib/glPopAttrib is not part of the core GL profile. Admittedly that cause a fair amount of bother, because the core profile friendly way would look something like this:
```c++
    struct BlendStatus
    {
      GLenum srcRGB;
      GLenum dstRGB;
      GLenum funcRGB;
      GLenum srcAlpha;
      GLenum dstAlpha;
      GLenum funcAlpha;
      bool enabled;
      void capture(uint32_t index);
      void set(uint32_t index) const;
    };

    void BlendStatus::capture(const uint32_t index)
    {
      glGetIntegeri_v(GL_BLEND_SRC_RGB, index, &srcRGB);
      glGetIntegeri_v(GL_BLEND_DST_RGB, index, &dstRGB);
      glGetIntegeri_v(GL_BLEND_EQUATION_RGB, index, &funcRGB);
      glGetIntegeri_v(GL_BLEND_SRC_ALPHA, index, &srcAlpha);
      glGetIntegeri_v(GL_BLEND_DST_ALPHA, index, &dstAlpha);
      glGetIntegeri_v(GL_BLEND_EQUATION_ALPHA, index, &funcAlpha);
      enabled = glIsEnabledi(GL_BLEND, index);
    }
    void BlendStatus::set(const uint32_t index) const
    {
      glBlendFuncSeparatei(index, srcRGB, dstRGB, srcAlpha, dstAlpha);
      glBlendEquationSeparatei(index, funcRGB, funcAlpha);
      if(enabled) glEnablei(GL_BLEND, index);
      else glDisablei(GL_BLEND, index);
    }
```
Then in place of push attrib:
```c++
    BlendStatus bs[8];
    for(uint32_t i = 0; i < 8; ++i)
      bs[i].capture(i);
```
And finally the pop attrib... 
```c++
    BlendStatus bs[8];
    for(uint32_t i = 0; i < 8; ++i)
      bs[i].capture(i);
```

Reply:
We have a similar class in an internal project that does all this for blending (though only for target 0), I'll adapt it for AL_USDMaya.